### PR TITLE
Add loading component

### PR DESCRIPTION
**Pull request by samaradel**
### Description

Replace the loading spinner with one global loading component

### Changes

image

### Related Issues

- [#2086](https://github.com/threefoldtech/tfgrid-sdk-ts/issues/2086)

### Checklist

- [ ] Tests included
- [x] Build pass
- [x] Documentation
- [x] Code format and docstrings
- [x] Screenshots/Video attached (needed for UI changes)

### CODE REVIEW

**Comment by MohamedElmdary** [comment]
@samaradel 

Can we define loading component globally as we will/are use/ing it in the entire app?
Also can we rename it to something like `TfLoading` to avoid conflicts?

to define component as global component in playground simple follow next steps
1. Add component type here https://github.com/threefoldtech/tfgrid-sdk-ts/blob/development/packages/playground/src/global-components.d.ts#L18-L31

2. Add component itself in global component array here https://github.com/threefoldtech/tfgrid-sdk-ts/blob/development/packages/playground/src/config.ts#L18-L31

3. use component where you need without importing it in every single component

**Comment by MohamedElmdary** [comment]
@samaradel 
alternate enhance might we can update VCircularProgress defaults in vuetify instance like here https://github.com/threefoldtech/tfgrid-sdk-ts/blob/development/packages/playground/src/plugins/vuetify.ts#L9-L13

so we can set `:size="70" :width="7" indeterminate color="primary"` as default

**Review by zaelgohary — CHANGES_REQUESTED** [review]
The following files should use the tfLoading as well.

https://github.com/threefoldtech/tfgrid-sdk-ts/blob/c3bd53df22d8680bba22c0faec6bcc1ace8746f1/packages/playground/src/components/app_info.vue#L15

https://github.com/threefoldtech/tfgrid-sdk-ts/blob/1ffaff6babbf6c8ff5477b198e2d11d2993deed7/packages/playground/src/components/dynamic_tabs.vue#L15

https://github.com/threefoldtech/tfgrid-sdk-ts/blob/816a2ba05a2f95ddf41e8a2009472b4dd1aaccdd/packages/playground/src/components/list_table.vue#L35-L42

https://github.com/threefoldtech/tfgrid-sdk-ts/blob/816a2ba05a2f95ddf41e8a2009472b4dd1aaccdd/packages/playground/src/components/list_table.vue#L62

https://github.com/threefoldtech/tfgrid-sdk-ts/blob/d88700a2343bfd5b90bf6959952823fbc032389e/packages/playground/src/components/node_resources_charts.vue#L22-L24

https://github.com/threefoldtech/tfgrid-sdk-ts/blob/d88700a2343bfd5b90bf6959952823fbc032389e/packages/playground/src/components/node_resources_charts.vue#L30

https://github.com/threefoldtech/tfgrid-sdk-ts/blob/37a495d15145ec49f4a2a5419fb48e40f8626024/packages/playground/src/dashboard/components/node_details.vue#L4

https://github.com/threefoldtech/tfgrid-sdk-ts/blob/5aa8c832901aafd834e05879419e274aa6766505/packages/playground/src/dashboard/components/user_nodes.vue#L56-L64

https://github.com/threefoldtech/tfgrid-sdk-ts/blob/73588ccad084695779f330362b0ae58da75b957e/packages/playground/src/views/farms.vue#L103

https://github.com/threefoldtech/tfgrid-sdk-ts/blob/99c08d5ed5c10ee603cb5bb8612bbe03a180cd7d/packages/playground/src/weblets/profile_manager.vue#L229

**Review by AlaaElattar — COMMENTED** [review]
- I think it needs to be at center of height.

[Screencast from 02-13-2024 02:38:49 PM.webm](https://github.com/threefoldtech/tfgrid-sdk-ts/assets/54944491/f3ec2cf6-c038-4909-ac56-5f79b65a050e)

**Comment by samaradel** [comment]
@AlaaElattar fixed

image

**Review by AlaaElattar — APPROVED** [review]
Great job

**Comment by zaelgohary** [comment]
@samaradel changes are still not applied in [here](https://github.com/threefoldtech/tfgrid-sdk-ts/pull/2140#pullrequestreview-1874659311)

**Comment by samaradel** [comment]
> <v-progress-circular

resolved

File: packages/playground/src/components/node_resources_charts.vue [hunk 1]
```diff
@@ -19,15 +19,14 @@
...
-          </v-progress-circular>
+          <tf-loading :model-value="item.value"> {{ item.value !== "NaN" ? item.value + "%" : "N/A" }}</tf-loading>
         </div>
       </div>
     </v-row>
 
     <v-row justify="center">
-      <v-progress-circular v-if="loading" indeterminate color="primary" :size="50" class="mt-10 mb-10" />
+      <tf-loading v-if="loading" class="mt-10 mb-10" />
+
```
Comment (zaelgohary) [inline packages/playground/src/components/node_resources_charts.vue:29]: Shouldn't we consider the size as well? I mean if the previous loading size was 50, then this also should be the same. This goes for all the loaders as well. Maybe we can create a size prop with a default size?

**Review by zaelgohary — CHANGES_REQUESTED** [review]

File: packages/playground/src/components/node_resources_charts.vue (hunk 1 above)
Comment (samaradel) [reply packages/playground/src/components/node_resources_charts.vue:29]: it's already default 50, if you check **packages/playground/src/plugins/vuetify.ts** you'll find the default props

**Comment by MohamedElmdary** [comment]
Loading not removed in node_details
https://github.com/threefoldtech/tfgrid-sdk-ts/assets/31689104/e3b426e8-4d7b-41bc-bbe0-e13cd4ea5d71

**Comment by MohamedElmdary** [comment]
> @MohamedElmdary I didn't remove any spinner, just replaced with one component

It's working in `development_314` but stopped work on this branch
I think you forgot to add v-if or something

File: packages/playground/src/components/node_resources_charts.vue
Comment (samaradel) [inline packages/playground/src/components/node_resources_charts.vue:1]: I think it a different issue , it just replaced with the component

**Comment by samaradel** [comment]
> > @MohamedElmdary I didn't remove any spinner, just replaced with one component
> 
> It's working in `development_314` but stopped work on this branch I think you forgot to add v-if or something

check  here please https://github.com/threefoldtech/tfgrid-sdk-ts/pull/2140/files#r1501783266

**Comment by samaradel** [comment]
> > > @MohamedElmdary I didn't remove any spinner, just replaced with one component
> > 
> > 
> > It's working in `development_314` but stopped work on this branch I think you forgot to add v-if or something
> 
> check here please https://github.com/threefoldtech/tfgrid-sdk-ts/pull/2140/files#r1501783266

fixed

**Comment by MohamedElmdary** [comment]
I don't think we should have our own loader component while using vuetify

**Comment by samaradel** [comment]
> I don't think we should have our own loader component while using vuetify

done

**Pull request merged**